src/DynamoUtilities/Option.cs

```diff
@@ -403,6 +403,16 @@ public bool Equals(_None<T> other)
             {
                 return this == other;
             }
+
+            public override bool Equals(object obj)
+            {
+                return Equals(obj as _None<T>);
```

Comment:
I hope null is handled somewhere downstream

Reply:
is handled in another function overload
![image](https://github.com/DynamoDS/Dynamo/assets/132492346/d9bf661b-242f-4d82-843b-1d681a2e6f7b)